packages/zql/src/query/query.ts

```diff
@@ -98,20 +102,36 @@ export type PullRow<TTable extends string, TSchema extends ZeroSchema> = {
   >;
 } & {};
 
+type RowNamespace<S extends Schema | TypeError> = S extends Schema
+  ? {
+      readonly [K in keyof S['tables'] &
+        string]: S['tables'][K] extends TableSchema
+        ? Row<S['tables'][K]>
+        : {
+            error: `The table schema for table ${K & string} you have registered with \`declare module '@rocicorp/zero'\` is incorrect.`;
+          };
+    }
+  : S;
+
 export type Row<
   T extends
+    | Schema
     | TableSchema
-    | Query<ZeroSchema, string, any>
-    | ((...args: any) => Query<ZeroSchema, string, any>),
-> = T extends TableSchema
-  ? {
-      readonly [K in keyof T['columns']]: SchemaValueToTSType<T['columns'][K]>;
-    }
-  : T extends
-        | Query<ZeroSchema, string, any>
-        | ((...args: any) => Query<ZeroSchema, string, any>)
-    ? QueryRowType<T>
-    : never;
+    | Query<string, ZeroSchema, any>
+    | ((...args: any) => Query<string, ZeroSchema, any>) = DefaultSchema,
+> = T extends Schema
```

Comment:
This type helper was updated to be able to support `Row['issue']` while being backwards compatible. This has actual autocompletion - parameterized types don't, like `Row<'issue'>`.

@aboodman WDYT about updating/simplifying these type helpers? We can easily add some kind of unified type helpers for mutators/queries. e.g. `MutatorParams`, `QueryParams`, `QueryResult`, (eventually?) `MutatorResult`.

Reply:
I like it but next release 😅